pkg/controllers/cassandra/nodepool/resource.go

```diff
@@ -194,6 +195,34 @@ func StatefulSetForCluster(
 									Name:  "HEAP_NEWSIZE",
 									Value: "100M",
 								},
+								// Deliberately set to a single space to force Cassandra to do a host name lookup.
+								// See https://github.com/apache/cassandra/blob/cassandra-3.11.2/conf/cassandra.yaml#L592
+								{
+									Name:  "CASSANDRA_LISTEN_ADDRESS",
+									Value: " ",
+								},
+								{
+									Name:  "CASSANDRA_BROADCAST_ADDRESS",
+									Value: " ",
+								},
+								{
+									Name:  "CASSANDRA_RPC_ADDRESS",
+									Value: " ",
+								},
+								// Set a non-existent default seed.
+								// The Kubernetes Seed Provider will fall back to a default seed host if it can't look up seeds via the CASSANDRA_SERVICE.
+								// And if the CASSANDRA_SEEDS environment variable is not set, it defaults to localhost.
+								// Which could cause confusion if a non-seed node is temporarily unable to lookup the seed nodes from the service.
```

Comment:
What would happen if we were to set this to localhost instead? Would the node fail to boot, or would it cause a split brain of some sort?

Reply:
I think it would mean that new nodes might consider themselves seeds in the event that the KubernetesSeedProvider plugin encounters an error.
So yeah, a split brain situation I think.